Makefile

```diff
@@ -1,4 +1,4 @@
-VERSION := $(shell git describe --tags --always)
+VERSION := $(subst v,,$(shell git describe --tags --abbrev=0))
```

Comment:
I'm not sure if commit suffix needs to be trimmed here.
Versions made from tag and another commit must have different version numbers.
This can be confusing.

I think we can trim this suffix in the check:
https://github.com/aquasecurity/trivy/blob/3c9eed53031efa7fa5448ba307df8ba38c53f2cc/pkg/commands/artifact/run.go#L471-L476

Reply:
>Versions made from tag and another commit must have different version numbers.

Could you elaborate on that?

Reply:
`subst` will remove any `v` in the tag. That would break if we have `v` elsewhere in the version like `v0.40v.6` for some reason. 

> It's no big deal, but it looks weird to me. v means version,so it is like Version: version 0.29.1.

We can't really trim it in the code as @knqyf263 suggested that the version during the compile itself should not have the `v`, otherwise it becomes `Version: version 0.29.1`.

Reply:
we can use `VERSION := $(patsubst v%,%,$(shell git describe --tags --abbrev=0))`

This will convert `v0.36v.0v` into `0.36v.0v`. Aka stripping `v` only from prefix.

Reply:
Oh I see what you mean now, yeah I'll put back the `--always`

Reply:
> Could you elaborate on that?

I only talked about the `--abbrev=0` flag.
At the moment when I build the binary I get `v<tag><commit number>` like this: `Version: v0.30.4-11-g44d53bed`
after the changes, we will only get the tag number for all commits between tags (for example, "Version: v0.30.4" for the actual commit in the main branch).

I think we only need to remove  `v` prefix.

Reply:
It makes sense. Good catch. Thanks.

Reply:
Please check now with the latest commit

| Case | Build                                                                                                  | versions                          | Output url                                                                                                                    |
| ---- | ------------------------------------------------------------------------------------------------------ | --------------------------------- | ----------------------------------------------------------------------------------------------------------------------------- |
| 1    | go build -ldflags "-s -w -X=main.version=v0.30.4" ./cmd/trivy/main.go                                  | Version: v0.30.4                  | Please see also https://aquasecurity.github.io/trivy/v0.30.4/docs/secret/scanning/#recommendation for faster secret detection |
| 2    | go build -ldflags "-s -w -X=main.version=0.30.4" ./cmd/trivy/main.go                                   | Version: 0.30.4                   | Please see also https://aquasecurity.github.io/trivy/v0.30.4/docs/secret/scanning/#recommendation for faster secret detection |
| 3    | go build ./cmd/trivy/main.go                                                                           | Version: dev                      | Please see also https://aquasecurity.github.io/trivy/dev/docs/secret/scanning/#recommendation for faster secret detection     |
| 4    | goreleaser check && goreleaser release --snapshot --rm-dist --skip-publish --skip-docker --skip-before | Version: 0.30.4-SNAPSHOT-f2b17e3c | Please see also https://aquasecurity.github.io/trivy/v0.30.4/docs/secret/scanning/#recommendation for faster secret detection |
| 5    | make build                                                                                             | Version: 0.30.4-6-gf2b17e3c       | Please see also https://aquasecurity.github.io/trivy/v0.30.4/docs/secret/scanning/#recommendation for faster secret detection |

Reply:
i think we can not use regex:
@Moulick what do you think of something like this?
```
	s = strings.TrimLeft(s, "v")
	r := strings.Split(s, "-")[0]
```

Reply:
@knqyf263 what about pre-release versions (e.g. `0.30.5-alpfa`). Should this be considered?

Reply:
I think we can just replace them with `dev` or accept the invalid URL.

Reply:
I was in favour of the regex as that would cover other cases for semver like if we had `v0.36.0+snapshot-1` or like `0.36.0-beta+exp.sha.5114f85`. wdyt?

Reply:
I agree with you. These versions require a regex, but as @knqyf263 said, we can return the invalid URL for pre-release versions and versions with build metadata.

Reply:
We can replace it with `dev`, but the question is again about detecting it's not a release version but a snapshot or pre-release or something, that again falls back to regex. I am the last person to want to use regex but also then I do want to produce valid URLs as much as possible

Reply:
In development, I don't mind even if an invalid URL is produced. If we want to make it valid as much as possible, we can use [go-version](https://github.com/aquasecurity/go-version/tree/main/pkg/semver).

```
ver := opts.AppVersion
if ver != "dev" {
    v, err := semver.Parse(opts.AppVersion)
    if err != nil {...}
   
    // Replace pre-release with "dev"
    // e.g. v0.36.0+snapshot-1
    if v.IsPreRelease || v.Metadata != "" {
        ver = "dev"
    } else {
        // Add "v" prefix, "0.31.0" => "v0.31.0"
        ver = "v" + ver
}
```

Reply:
Sure, I'll switch to `go-version`.

Reply:
resolved

Reply:
As mentioned earlier, we want to use our go-version, not hashicorp/go-version.
https://github.com/aquasecurity/trivy/pull/2607#discussion_r936372886